core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.controller.spec.ts

```diff
@@ -457,4 +457,135 @@ describe('Translation Suggestion Review Modal Controller', function() {
+      change: {
+        content_id: 'hint_1',
+        content_html: 'Translation',
+        translation_html: 'Tradução',
+        state_name: 'StateName'
+      },
+      exploration_content_html: 'Translation'
+    };
+
+    const contribution1 = {
+      suggestion: suggestion1,
+      details: null
+    };
+    const contribution2 = {
+      suggestion: suggestion2,
+      details: {
+        topic_name: 'topic_2',
+        story_title: 'story_2',
+        chapter_title: 'chapter_2'
+      }
+    };
+
+    const deletedContribution = {
+      suggestion: suggestion3,
+      details: null
+    };
+
+    const suggestionIdToContribution = {
+      suggestion_1: contribution1,
+      suggestion_deleted: deletedContribution,
+      suggestion_2: contribution2
+    };
+
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      const $rootScope = $injector.get('$rootScope');
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+
+      $scope = $rootScope.$new();
+      $controller('TranslationSuggestionReviewModalController', {
+        $scope: $scope,
+        $uibModalInstance: $uibModalInstance,
+        initialSuggestionId: 'suggestion_1',
+        subheading: subheading,
+        reviewable: reviewable,
+        suggestionIdToContribution: angular.copy(suggestionIdToContribution)
+      });
+    }));
+    it(
+      'should reject suggestion in suggestion modal service when clicking ' +
+      'on reject and review next suggestion button', function() {
+        expect($scope.activeSuggestionId).toBe('suggestion_1');
+        expect($scope.activeSuggestion).toEqual(suggestion1);
+        expect($scope.reviewable).toBe(reviewable);
+        expect($scope.reviewMessage).toBe('');
+
+        spyOn(contributionAndReviewService, 'resolveSuggestionToExploration')
+          .and.callFake((
+              targetId, suggestionId, action, reviewMessage, commitMessage,
+              callback) => {
+            callback();
+          });
+        spyOn(
+          SiteAnalyticsService,
+          'registerContributorDashboardRejectSuggestion');
+
+        $scope.reviewMessage = 'Review message example';
+        $scope.rejectAndReviewNext();
+
+        expect($scope.activeSuggestionId).toBe('suggestion_2');
+        expect($scope.activeSuggestion).toEqual(suggestion2);
+        expect($scope.reviewable).toBe(reviewable);
+        expect($scope.reviewMessage).toBe('');
+        expect(
+          SiteAnalyticsService.registerContributorDashboardRejectSuggestion)
+          .toHaveBeenCalledWith('Translation');
+        expect(contributionAndReviewService.resolveSuggestionToExploration)
+          .toHaveBeenCalledWith(
+            '1', 'suggestion_1', 'reject', 'Review message example',
+            'hint section of "StateName" card', $scope.showNextItemToReview);
+
+        $scope.reviewMessage = 'Review message example 2';
+        $scope.rejectAndReviewNext();
```

Comment:
Why is this being executed again?

Reply:
Are you asking about why the it function is duplicated down here? If so it is because I wanted to test if the reject and accept process for reviewers can handle a `suggestionIdToContribution` with deleted contributions.

Reply:
Sorry, for not being clear. i was actually asking why ` $scope.rejectAndReviewNext();` is being executed twice.

Reply:
Ah okay, So the first `$scope.rejectAndReviewNext() `rejects suggestion 1 and loads suggestion 2. The second `$scope.rejectAndReviewNext() ` rejects suggestion 2 and attempts to load deletedSuggestion and finds that it is empty and closes the modal

Reply:
Can you split it into 2 separate tests?

Reply:
I can't because I need to reject twice to reach the deleted contribution. I could remove contribution2 if you want to remove one of the reject calls.

Reply:
You could either put a comment explaining why we need to reject twice or remove one of the reject calls. I'll leave this up to your discretion.

Reply:
> remove one of the reject calls.

I chose to do this PTAL!